platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/verification/ChecksumAndSignatureVerificationOverride.java

```diff
@@ -74,9 +75,12 @@ public class ChecksumAndSignatureVerificationOverride implements DependencyVerif
     private final Set<VerificationQuery> verificationQueries = Sets.newConcurrentHashSet();
     private final Deque<VerificationEvent> verificationEvents = Queues.newArrayDeque();
     private final AtomicBoolean closed = new AtomicBoolean();
-    private final AtomicBoolean hasFatalFailure = new AtomicBoolean();
     private final DependencyVerificationReportWriter reportWriter;
 
+    // Must hold lock on `failures` to access `failures` or `hasFatalFailure`
```

Comment:
Can we change this so it's not synchronizing on a third-party class? It's usually a bad idea to rely on a class not using it's lock somewhere.